contracts/starknet/src/oracle/library.cairo

```diff
@@ -279,10 +316,54 @@ namespace Oracle {
         return (entries_len, entries, last_updated_timestamp);
     }
 
-    func get_spot_entry{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        pair_id: felt, source
-    ) -> (entry: SpotEntry) {
-        let (entry) = Oracle_spot_entry_storage.read(pair_id, source);
+    func get_spot_entry{
+        bitwise_ptr: BitwiseBuiltin*,
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+    }(pair_id: felt, source) -> (entry: SpotEntry) {
+        alloc_locals;
+        let (_entry) = Oracle_spot_entry_storage.read(pair_id, source);
+
+        let timestamp = actual_get_element_at(_entry.timestamp__volume__price, 0, 31);
+        let volume = actual_get_element_at(_entry.timestamp__volume__price, 32, 42);
+        let price = actual_get_element_at(_entry.timestamp__volume__price, 75, 128);
```

Comment:
Timestamp has 32 bits which works well. Volume only has 11 bits which seems to me like is too little. Also seems like we are leaving 123 (251-128) bits unused?

Reply:
no, it has 42.  2 ** 42 is 4 trillion, i think that's fine

Reply:
why do we need to use all the bits if there's no need for them @jonasalexander ?

Reply:
I'd recommend reading the code for `actual_get_element_at` to familiarize yourself with how the functions work.